mpr-research-data.py

```diff
@@ -139,56 +119,156 @@ def sliceAndPushToGCP(courseDF, retrieveDF, client, targetBucketName=targetBucke
         try:
             logging.info(f'Slicing: {outputFilename}')
             saveDF = retrieveDF[retrieveDF['CourseID'] == courseID]
-            saveDF.to_csv(outputFilename, sep="\t", quoting=3,
-                          quotechar="", escapechar="\\")
-
-        except Exception as e:
-            logging.error(f'Error Message: {e}')
-            logging.error(f'Failed to save Course Data for {outputFilename}.')
-            continue
-
-        try:
             logging.info(f'Saving to GCP: {outputFilename}')
             blob = bucket.blob(outputFilename)
-            blob.upload_from_filename(outputFilename)
-            os.remove(outputFilename)
-        except Exception as e:
+            blob.upload_from_string(saveDF.to_csv(sep="\t", quoting=3, quotechar="", escapechar="\\"), 'text/tsv')
```

Comment:
Is this the main line(s) involved in "Changed how files are saved before GCP upload"?  Because of the way a lot of things got mixed into this commit, I wasn't sure.

I'd noticed this problem near the end of the workday today, just before I suggested resolving the unsigned commits.  I thought there could be trouble with saving the TSV data to files and reading it back because ➊ the container probably has a read-only filesystem and ➋ no path like `/tmp` was used for the files.

I was going to open a new issue to deal with that problem after this PR was merged.  You've already handled it in this PR, and that's good, too.

Reply:
This popped up as an issue when I was testing the Jenkins config today. Hence the change.

Reply:
✅